File: python-code/ex6/carro.py
class Motor:
  def __init__(self, tipo):
    self.tipo = tipo

  def exibir_tipo(self):
    return f"Tipo de motor: {self.tipo}"

class Carro:
  def __init__(self, marca, modelo, motor):
    self.marca = marca
    self.modelo = modelo
    self.motor = motor

  def exibir_detalhes(self):
    return f"Marca: {self.marca}, Modelo: {self.modelo}, {self.motor.exibir_tipo()}"

motor = Motor("V8")
carro = Carro("Volswagen", "Golf", motor)

print(carro.exibir_detalhes())

File: java-code/src/main/java/ex6/Carro.java
package ex6;

public class Carro {
    private String marca;
    private String modelo;
    private Motor motor;

    public Carro(String marca, String modelo, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.motor = motor;
    }

    public String exibirDetalhes() {
        return "Marca: " + marca + ", Modelo: " + modelo + ", " + motor.exibirTipo();
    }

    public static void main(String[] args) {
        Motor motor = new Motor("V8");
        Carro carro = new Carro("Ford", "Mustang", motor);

        System.out.println(carro.exibirDetalhes());
    }
}
